### ISSUE TITLE
Adversarial loss not counted by model's optimizer

### ISSUE THREAD

**Issue by tudragon154203** [MyeongJin-Kim/Learning-Texture-Invariant-Representation#11]
Hi MyeongJin-Kim, 

In train_gta2cityscapes.py I notice that this line:
`# optimize
optimizer.step()`
is called before the adversarial loss's gradients are calculated:
`loss_adv = loss_adv * args.lambda_adv
loss_adv.backward()`
I believe that implies the loss_adv is not used by the optimizer to update the model's params.

Can you explain more and should I put optimizer.step() after loss_adv.backward()?
